File: PetCareManager/src/PetCareManager.Core/Model/MedicationAggregate/Medication.cs
// Licensed to the .NET Foundation under one or more agreements.
// The .NET Foundation licenses this file to you under the MIT license.

namespace PetCareManager.Core;

public class Medication
{
    public Guid MedicationId { get; set; }

    /// <summary>
    /// Gets or sets the tenant ID for multi-tenant isolation.
    /// </summary>
    public Guid TenantId { get; set; }
    public Guid PetId { get; set; }
    public Pet? Pet { get; set; }
    public string Name { get; set; } = string.Empty;
    public string? Dosage { get; set; }
    public string? Frequency { get; set; }
    public DateTime? StartDate { get; set; }
    public DateTime? EndDate { get; set; }
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
}

File: PetCareManager/src/PetCareManager.WebApp/projects/pet-care-manager/src/app/models/medication.ts
export interface Medication {
  medicationId: string;
  petId: string;
  name: string;
  dosage?: string;
  frequency?: string;
  startDate?: string;
  endDate?: string;
  createdAt: string;
}

export interface CreateMedicationDto {
  petId: string;
  name: string;
  dosage?: string;
  frequency?: string;
  startDate?: string;
  endDate?: string;
}

export interface UpdateMedicationDto {
  medicationId: string;
  petId: string;
  name: string;
  dosage?: string;
  frequency?: string;
  startDate?: string;
  endDate?: string;
}
